### PR TITLE
Allow config updates to Admins & CAs

### PR DESCRIPTION
**Pull request by tock-ibm**
Dynamic config of ClusterConfig is currently not supported and causes panic.
This commit allows config changes to ClusterConfig.Admins and ClusterConfig.CertAuthConfig,
as those have no effect on the replication component.

The server will panic if ClusterConfig.Nodes or ClusterConfig.ConsensusConfig are changed.
This will change once dynamic reconfig of consensus related objects is implemented.

Signed-off-by: Yoav Tock <tock@il.ibm.com>

### CODE REVIEW

File: internal/replication/blockreplicator.go [hunk 1]
```diff
@@ -309,8 +309,11 @@ func (br *BlockReplicator) deliverEntries(committedEntries []raftpb.Entry) bool
 			}
 
 			if reConfig != nil {
-				// TODO support reconfig
-				br.lg.Panic("Unexpected update to ClusterConfig during normal entry")
+				clusterConfig := reConfig.(*types.ClusterConfig)
+				if err := br.updateClusterConfig(clusterConfig); err != nil {
+					// TODO support dynamic re-config
+					br.lg.Panicf("Unexpected update to ClusterConfig during normal entry: error: %s", err)
```
Comment (gennadylaventman) [inline internal/replication/blockreplicator.go:315]: Can you rephrase log message?

File: internal/replication/env_test.go [hunk 2]
```diff
@@ -36,6 +37,13 @@ var clusterConfig1node = &types.ClusterConfig{
 		Port:        nodePortBase + 1,
 		Certificate: []byte("bogus-cert"),
 	}},
+	Admins: []*types.Admin{&types.Admin{
+		Id:                   "admin",
+		Certificate:          []byte("something"),
+		XXX_NoUnkeyedLiteral: struct{}{},
+		XXX_unrecognized:     nil,
+		XXX_sizecache:        0,
```
Comment (gennadylaventman) [inline internal/replication/env_test.go:45]: Should we fill XXX_ fileds?

File: internal/replication/utils.go [hunk 3]
```diff
@@ -25,3 +27,41 @@ func raftEntryString(e raftpb.Entry) string {
...
 		e.Term, e.Index, e.Type, len(e.Data), h.Sum64())
 }
+
+func ClassifyClusterReConfig(currentConfig, updatedConfig *types.ClusterConfig) (nodes bool, consensus bool, ca bool, admins bool) {
+	curNodes := currentConfig.GetNodes()
+	updNodes := updatedConfig.GetNodes()
+	if len(curNodes) != len(updNodes) {
+		nodes = true
+	} else {
+		for i, node := range curNodes {
```
Comment (gennadylaventman) [inline internal/replication/utils.go:37]: Do we keep nodes order in list? Can't it be same nodes, but in different order?

Comment (cendhu) [reply internal/replication/utils.go:37]: Yes, good point.

File: internal/replication/blockreplicator.go (hunk 1 above)
Comment (tock-ibm) [reply internal/replication/blockreplicator.go:315]: I hope this is better"
`br.lg.Panicf("Failed to update to ClusterConfig during raft normal entry: error: %s", err)`

Anyway, this log message is temporary until dynamic consensus config is implemented...

File: internal/replication/env_test.go (hunk 2 above)
Comment (tock-ibm) [reply internal/replication/env_test.go:45]: removed

File: internal/replication/utils.go (hunk 3 above)
Comment (tock-ibm) [reply internal/replication/utils.go]: Done, for both nodes and admins.

**Review by gennadylaventman — APPROVED** [review]
